course5/homework06/test.h

```diff
@@ -0,0 +1,103 @@
+#include <cstring>
+
+template <typename T>
+class TestClass
+{
+public:
+    void test_const_char_constructor()
+    {
+        auto str = "this is string with length 29";
+        auto s = T(str);
+        assert(s == "this is string with length 29");
+    }
+
+    void test_move_constructor()
+    {
+        auto str = "this is string with length 29";
+        auto s = T(str);
+        auto s2 = std::move(s);
+        assert (s2 == "this is string with length 29");
```

Comment:
move only casts normal reference into r-value reference. It's up to the author of the class (you!) to ensure that when rvalue ref (```&&```) is ued, you actually move.
You should test it. One way you can so it is to compare the value returned by ```data()``` or ```c_str()```: s.c_str() before move should return the same as s2.c_str() after move. The same address. This would mean that s2 is using the same buffer s used to use.